### PR TITLE
Implement `Display` for `TokenKind`

### PR DESCRIPTION
**Pull request by dhruvmanila**
## Summary

This PR implements `Display` for `TokenKind` to improve error messages.

### CODE REVIEW

File: crates/ruff_python_parser/src/token.rs [hunk 1]
```diff
@@ -900,6 +900,120 @@ impl TryFrom<TokenKind> for UnaryOp {
     }
 }
 
+impl fmt::Display for TokenKind {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let value = match self {
+            TokenKind::Unknown => "<Unknown>",
+            TokenKind::StartModule => "StartProgram",
```
Comment (MichaReiser) [inline crates/ruff_python_parser/src/token.rs:907]: Is it intentional that the display text is different from the token kind?

**Review by MichaReiser — COMMENTED** [review]
What's the motivation for implementing `Display` on `TokenKind`? Is it for better error messages? 

I'm asking because I agree that showing `.` instead of `Dot` is better, but I'm not sure if we have good display names for some of the tokens (e.g. `Name`, `Int`) and I would consider these names to be internal but not public terminology. 

Would it make sense to have a `as_str()` method that returns `Option<&str>` instead? Or is this not sufficient in case of the *Expected `x`, but at `y`* error message?

File: crates/ruff_python_parser/src/token.rs (hunk 1 above)
Comment (dhruvmanila) [reply crates/ruff_python_parser/src/token.rs:907]: No, just an oversight. I'll fix this.

**Comment by dhruvmanila** [comment]
@MichaReiser 

> What's the motivation for implementing `Display` on `TokenKind`? Is it for better error messages?

Yes

> I'm asking because I agree that showing `.` instead of `Dot` is better, but I'm not sure if we have good display names for some of the tokens (e.g. `Name`, `Int`) and I would consider these names to be internal but not public terminology.
>
> Would it make sense to have a `as_str()` method that returns `Option<&str>` instead? Or is this not sufficient in case of the _Expected `x`, but at `y`_ error message?

Yeah, that could be done, we would need to fallback to the Debug implementation for the remaining tokens.

**Pull request merged**